### PR TITLE
Updates Discount Functions sample app to new template

### PR DESCRIPTION
**Pull request by marlonmarcello**
## Background

This PR updates the Discount Functions Sample App to the new node template and new standards.  
Instead of simply removing the `gql` proxy I decided to go ahead and update the whole sample app to the new node template, that way all 3 sample apps will look and behave very similarly.  
Fixes [24](https://github.com/orgs/Shopify/projects/3489/views/24).  

## 🎩
To top hat, please clone the repo, checkout the working branch (`modernize-discount-sample-app`) and run it on your development store with `npm run deploy --reset; npm run dev`

### CODE REVIEW

File: sample-apps/discount-functions-sample-app/web/index.js [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+  SCOPES: process.env.SCOPES.split(","),
+  HOST_NAME: process.env.HOST.replace(/https?:\/\//, ""),
+  HOST_SCHEME: process.env.HOST.split("://")[0],
+  API_VERSION: LATEST_API_VERSION,
+  IS_EMBEDDED_APP: true,
+  // This should be replaced with your preferred storage strategy
+  // See note below regarding using CustomSessionStorage with this template.
+  SESSION_STORAGE: new Shopify.Session.SQLiteSessionStorage(DB_PATH),
+  ...(process.env.SHOP_CUSTOM_DOMAIN && {
+    CUSTOM_SHOP_DOMAINS: [process.env.SHOP_CUSTOM_DOMAIN],
```
Comment (mssalemi) [inline sample-apps/discount-functions-sample-app/web/index.js:45]: Just a question, but does SPIN update this env here? Or do we need to set that?

Comment (andrewhassan) [reply sample-apps/discount-functions-sample-app/web/index.js:45]: I wouldn't expect this to be populated by Spin, but we can confirm by printing out the value in your instance: `echo $SHOP_CUSTOM_DOMAIN`

File: sample-apps/discount-functions-sample-app/web/index.js [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+    const { status, data } = await queryResponse(req, res, query, reducer);
+
+    return res.status(status).send(data);
+  });
+
+  app.put("/api/discount/:id", async (req, res) => {
+    const payload = req.body;
+    const gid = idToGid(req.params.id);
+
+    console.log({ payload, gid });
```
Comment (andrewhassan) [inline sample-apps/discount-functions-sample-app/web/index.js:248]: Nit: I think we should remove this console.log.

File: sample-apps/discount-functions-sample-app/web/index.js (hunk 1 above)
Comment (marlonmarcello) [reply sample-apps/discount-functions-sample-app/web/index.js:45]: Forgot to mention that you need to run that inside the `cli` repo, like you normally do when setting up functions
https://vault.shopify.io/page/Using-Functions-in-Spin~s7J5.md

File: sample-apps/discount-functions-sample-app/web/index.js [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+
+    // update metafield
```
Comment (andrewhassan) [inline sample-apps/discount-functions-sample-app/web/index.js:250]: Nit: This comment probably isn't necessary (and it should refer to discount).

**Review by andrewhassan — APPROVED** [review]
Okay I got my CLI to work by bumping the version to `3.21.0`! I think you should bump it in this PR too.

Otherwise, LGTM aside from a couple nits.

**Pull request merged**